### PR TITLE
Refactor Status to avoid large stack size in happy path

### PR DESCRIPTION
**Pull request by PragmaTwice**
Follow the implementation of Status in RocksDB, google absl, doris and more, we refactor the Status to make it just a pointer (8 instead of 40 for libstdc++ std::string).

For happy path (OK status) with large probability, it benefits the performance.

### CODE REVIEW

**Review by torwig — DISMISSED** [review]
LGTM.
Good job, @PragmaTwice!

File: src/common/status.h [hunk 1]
```diff
@@ -67,29 +66,41 @@ class [[nodiscard]] Status {
...
+    return impl_->msg_;
   }
 
   std::string Msg() && {
     if (*this) return ok_msg;
-    return std::move(msg_);
+    return std::move(impl_->msg_);
   }
 
   static Status OK() { return {}; }
```
Comment (tisonkun) [inline src/common/status.h:106]: Shall we return `cOK` here also to avoid new object instance?

Comment (tisonkun) [reply src/common/status.h:106]: Since we don't return a reference here, it should not be different.

**Review by tisonkun — DISMISSED** [review]

**Comment by tisonkun** [comment]
There are still ICE:

```
lto1: internal compiler error: in add_symbol_to_partition_1, at lto/lto-partition.c:153
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-9/README.Bugs> for instructions.
lto-wrapper: fatal error: /usr/bin/g++ returned 1 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
```

**Comment by PragmaTwice** [comment]
> There are still ICE:
> 
> ```
> lto1: internal compiler error: in add_symbol_to_partition_1, at lto/lto-partition.c:153
> Please submit a full bug report,
> with preprocessed source if appropriate.
> See <file:///usr/share/doc/gcc-9/README.Bugs> for instructions.
> lto-wrapper: fatal error: /usr/bin/g++ returned 1 exit status
> compilation terminated.
> /usr/bin/ld: error: lto-wrapper failed
> collect2: error: ld returned 1 exit status
> ```

sad

**Review by git-hulk — APPROVED** [review]

**Comment by PragmaTwice** [comment]
Will split this PR to multiple ones against ICE.

**Pull request closed without merging**